NewmanPostman/newmantask.ts

```diff
@@ -2,50 +2,82 @@ import path = require('path');
-
-        let sslClientCert = tl.getPathInput('sslClientCert', false, true);
-        newman.argIf(typeof sslClientCert != 'undefined' && sslClientCert, ['--ssl-client-cert', sslClientCert]);
-
-        let sslClientKey = tl.getPathInput('sslClientKey', false, true);
-        newman.argIf(typeof sslClientKey != 'undefined' && sslClientKey, ['--ssl-client-key', sslClientKey]);
-
-        let reporterHtmlTemplate = tl.getPathInput('reporterHtmlTemplate', false, true);
-        newman.argIf(typeof reporterHtmlTemplate != 'undefined' && reporterHtmlTemplate, ['--reporter-html-template', reporterHtmlTemplate]);
-
-        let reporterHtmlExport = tl.getInput('reporterHtmlExport');
-        newman.argIf(typeof reporterHtmlExport != 'undefined' && reporterHtmlExport, ['--reporter-html-export', reporterHtmlExport]);
+        let collectionFileSource = tl.getPathInput('collectionFileSource', true, true);
+        var taskSuccess = true;
+        if (tl.stats(collectionFileSource).isDirectory()) {
+            let contents: string[] = tl.getDelimitedInput('Contents', '\n', true);
+            collectionFileSource = path.normalize(collectionFileSource);
 
-        let reporterJsonExport = tl.getInput('reporterJsonExport');
-        newman.argIf(typeof reporterJsonExport != 'undefined' && reporterJsonExport, ['--reporter-json-export', reporterJsonExport]);
+            let allPaths: string[] = tl.find(collectionFileSource);
+            let matchedPaths: string[] = tl.match(allPaths, contents, collectionFileSource);
+            let matchedFiles: string[] = matchedPaths.filter((itemPath: string) => !tl.stats(itemPath).isDirectory());
 
-        let reporters = tl.getInput('reporters');
-        newman.argIf(typeof reporters != 'undefined' && reporters, ['-r', reporters]);
+            console.log("found %d files", matchedFiles.length);
 
-        let delayRequest = tl.getInput('delayRequest');
-        newman.argIf(typeof delayRequest != 'undefined' && delayRequest, ['--delay-request', delayRequest]);
+            if (matchedFiles.length > 0) {
+                matchedFiles.forEach((file: string) => {
+                    var newman: trm.ToolRunner = GetToolRunner();
+                    newman.arg('run');
+                    newman.arg(file);
+                    newman.arg(['-e', tl.getPathInput('environment', true, true)]);
 
-        let timeoutRequest = tl.getInput('timeoutRequest');
-        newman.argIf(typeof timeoutRequest != 'undefined' && timeoutRequest, ['--timeout-request', timeoutRequest]);
+                    var execResponse = newman.execSync();
 
-        let numberOfIterations = tl.getInput('numberOfIterations');
-        newman.argIf(typeof numberOfIterations != 'undefined' && numberOfIterations, ['-n', numberOfIterations]);
+                    if (execResponse.code === 1) {
+                        taskSuccess = false;
+                    }
+                });
+            }
+            else {
+                taskSuccess = false;
```

Comment:
I don't have a ton of knowledge around the "vsts-task-lib/task" (or the tl variable), but seems that we should either somehow track the problem and report it when it fails or otherwise, if we can, set the result right here. However, we'd still want to aggregate the errors.

Reply:
So, I didn't make any changes to how this is currently set up for these reasons:

1. I added logging to lines 55 (for when a single collection fails) and 61 (whine no collections files are found) so it will be reported in the logs for the release/build.
1. I could fail the task earlier (like on line 56), but if the user has multiple collections files, they may still want the others to run before completely failing the task.

I suppose I could create an array to store the execResponse for each collection run, then log the errors at the end, but that seems like a little bit of overkill... I'll leave that up to you though... just make the call and I'll make the change :)